src/Catalyst.Core.Modules.Dfs.Tests/IntegrationTests/CoreApi/DagApiTest.cs

```diff
@@ -0,0 +1,161 @@
+            Assert.NotNull(id);
+            Assert.Equal(expectedId, (string) id);
+
+            var actual = await ipfs.DagApi.GetAsync(id);
+            Assert.NotNull(actual);
+            Assert.Equal(expected["a"], actual["a"]);
+
+            var value = (string) await ipfs.DagApi.GetAsync(expectedId + "/a");
+            Assert.Equal(expected["a"], value);
+        }
+
+        [Fact]
+        public async Task PutAndGet_poco()
+        {
+            var expected = new name {first = "John", last = "Smith"};
+            var id = await ipfs.DagApi.PutAsync(expected);
+            Assert.NotNull(id);
+
+            var actual = await ipfs.DagApi.GetAsync<name>(id);
+            Assert.NotNull(actual);
+            Assert.Equal(expected.first, actual.first);
+            Assert.Equal(expected.last, actual.last);
+
+            var value = (string) await ipfs.DagApi.GetAsync(id.Encode() + "/last");
+            Assert.Equal(expected.last, value);
+        }
+
+        [Fact]
+        public async Task PutAndGet_poco_CidEncoding()
+        {
+            var expected = new name {first = "John", last = "Smith"};
+            var id = await ipfs.DagApi.PutAsync(expected, encoding: "base32");
+            Assert.NotNull(id);
+            Assert.Equal("base32", id.Encoding);
+            Assert.Equal(1, id.Version);
+
+            var actual = await ipfs.DagApi.GetAsync<name>(id);
+            Assert.NotNull(actual);
+            Assert.Equal(expected.first, actual.first);
+            Assert.Equal(expected.last, actual.last);
+
+            var value = (string) await ipfs.DagApi.GetAsync(id.Encode() + "/last");
+            Assert.Equal(expected.last, value);
+        }
+
+        [Fact]
+        public async Task PutAndGet_POCO()
+        {
+            var expected = new Name {First = "John", Last = "Smith"};
+            var id = await ipfs.DagApi.PutAsync(expected);
+            Assert.NotNull(id);
+
+            var actual = await ipfs.DagApi.GetAsync<Name>(id);
+            Assert.NotNull(actual);
+            Assert.Equal(expected.First, actual.First);
+            Assert.Equal(expected.Last, actual.Last);
+
+            var value = (string) await ipfs.DagApi.GetAsync(id.Encode() + "/Last");
+            Assert.Equal(expected.Last, value);
+        }
+
+        [Fact]
+        public async Task Get_Raw2()
+        {
+            var data = Encoding.UTF8.GetBytes("abc");
+            var id = await ipfs.BlockApi.PutAsync(data, "raw");
+            Assert.Equal("bafkreif2pall7dybz7vecqka3zo24irdwabwdi4wc55jznaq75q7eaavvu", id.Encode());
+
+            var actual = await ipfs.DagApi.GetAsync(id);
+            Assert.Equal(Convert.ToBase64String(data), (string) actual["data"]);
+        }
+
+        // // https://github.com/ipfs/interface-ipfs-core/blob/master/SPEC/DAG.md
+        // [Fact]
+        // [Ignore("https://github.com/richardschneider/net-ipfs-engine/issues/30")]
+        // public async Task Example1()
+        // {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)